### ISSUE TITLE
[Bug] all sites are blank since 84.1.1 on Adreno 510

### ISSUE THREAD

**Issue by titiger** [mozilla-mobile/fenix#17181]
Firefox does not show(render) any website. No google search, nothing. You just got a white or black pane where content should be shown, no matter which URL. Looks like others have this problem too:  https://www.reddit.com/r/firefox/comments/kfg3zb/8411_stable_any_url_only_shows_a_blank_background/.
It looks like you can still click on unvisible links on this blank screen and the url changes, so it must be the rendering .

I am using firefox for a long time on this device and never got a ( serious ;-) )problem before. Just V 84.1.1 is completly broken for me.
I have no plugins installed and already restarted the device and so on...

## Steps to reproduce
I got an update from  playstore and  now firefox does not show any more webites. You either get a white or a black screen.
Just install firefox 84.1.1 and you get the trouble. ( on this device )
### Expected behavior
firefox should show websites

### Actual behavior
firefox just shows white or black blank screen as content.
### Device information
* Android device: Xiaomi redmi note 3 global
 Android 6.0.1 MMB29M ( MIUMI Global 10.2 Stable 10.2.1.0(MHRMIXXM))

* Fenix version: 84.1.1 (Build  #2015781795)

**Comment by sheikh-azharuddin** [comment]
Works fine here...  Reinstall and check once

**Comment by Mikhail-tech** [comment]
++++++++++++++++++++++++++++++++++++++
2 weeks ago, I updated my Firefox Nightly (before i dont update for about 2 months, and everything was fine) and he just stopped loading ALL the sites normally - the page loads completely, but the screen remains white (or black). I can't even log in to my Firefox account (for the same reason). I tried to clean the cache, disable add-ons, restart the device, reinstall the application (about 10 times after each update) - nothing helped, and updates from the GP were not fixed that. Then I went to the GP and found out that there is Firefox Daylight. I downloaded it - everything works fine (version 83.1). I tried to download Firefox Beta - the same problem as with Firefox Nightly. Well, I think: okay, I will continue to work with Firefox Daylight (although there were some features that I liked in Firefox Nightly). But a week ago I updated Firefox Daylight and the same problem started. In General, it all started on all Firefox browsers after updates with the addition of the function of displaying tabs a grid (this problem is not specifically related to this function, it just started after it was added, and what was actually added in addition to this, which so greatly affected the browser's operation - I do not know). Leeco Cool 1 smartphone, Android 6.0.1, ARM64-v8a, 3 GB of RAM. I have already written to various forums, but no one has given an answer so far. I apologize in advance for the mistakes, I wrote through the translator

**Comment by xM8WVqaG** [comment]
I am also having this issue, I have tested a few older versions and it looks like the problem started with `84.1.0`.

**Device:** Xiaomi Redmi Note 3 Pro
**OS:** Lineage 13.0 (Android 6.0.1)

| Version | APK Source | Working? |
| --- | --- | --- |
| 85.0.0-beta.1 | [Play Store](https://play.google.com/store/apps/details?id=org.mozilla.firefox_beta) | blank screen |
| 84.1.1 | [Play Store](https://play.google.com/store/apps/details?id=org.mozilla.firefox) | blank screen |
| 84.1.0 | [GitHub](https://github.com/mozilla-mobile/fenix/releases/tag/v84.1.0) | blank screen |
| 83.1.0 | [GitHub](https://github.com/mozilla-mobile/fenix/releases/tag/v83.1.0) | working |

With a fresh install of `85.0.0-beta.1` from the Play Store I've captured and a logcat and a video.

Logcat filtered to `org.mozilla.firefox_beta` and `org.mozilla.firefox_beta:tab0`:

[demo-filtered.log](https://github.com/mozilla-mobile/fenix/files/5730492/demo-filtered.log)

Video attempting to load Wikipedia:

https://user-images.githubusercontent.com/44505331/102901575-b4c30f00-4465-11eb-82b7-c1f117ec7b8c.mp4

NOTE, although the window appears blank, the actual DOM does appear to have been rendered as you can see me interacting with hyperlinks and highlighting text by long pressing blindly.

Another thread of people also with this issue: https://support.mozilla.org/en-US/questions/1318535

**Comment by Mikhail-tech** [comment]
I have the same problem (exactly). Apparently this is due to the Android version (6.0.1). Since everyone who writes about this is united by the version of Android. Although it fits the system requirements...

**Comment by kbrosnan** [comment]
I am able to browse websites using a Spice Mi-498 (Mali-400MP2) running Android 6.0. 

Looking at the phones that are experiencing this they are all running Adreno 510. Going to move this to Bugzilla as this is a Graphics issue.

**Comment by kbrosnan** [comment]
Moved to bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1683936

Change performed by the [Move to Bugzilla add-on](https://addons.mozilla.org/en-US/firefox/addon/move-to-bugzilla/).

**Comment by kbrosnan** [comment]
If anyone who experiences this has beta or nightly. Are you about to visit `about:config` and set `gfx.webrender.force-disabled` to true? Though I doubt this is possible given that about:config is a Gecko rendered page.

**Comment by DiegoARL38** [comment]
@kbrosnan  They could try using TalkBack to do it.

**Comment by Mikhail-tech** [comment]
@kbrosnan, @DiegoARL38, i tried changing the value of `gfx.webrender.force-disabled` to `true` using TalkBack - nothing changed. What's going to happen?

**Comment by Mikhail-tech** [comment]
Oooooooooooooooo, restarted the browser - its worked! It's alive!

**Comment by jamienicol** [comment]
@titiger  @Mikhail-tech or anyone else who is affected by this: could you please test a couple of apks and see if the problem is fixed.

If your android OS is 32 bit:
This apk should be broken: https://firefox-ci-tc.services.mozilla.com/api/queue/v1/task/fQNzULF0R0ymouVBur1jDA/runs/0/artifacts/public/build/geckoview_example.apk
This apk should hopefully fix the problem: https://firefox-ci-tc.services.mozilla.com/api/queue/v1/task/JUyhA--8Rmy008a4pVTXZw/runs/0/artifacts/public/build/geckoview_example.apk

If your android is 64 bit:
This apk should be broken: https://firefox-ci-tc.services.mozilla.com/api/queue/v1/task/VYds3sRoSbq23ZdDjEBS7w/runs/0/artifacts/public/build/geckoview_example.apk
This apk should hopefully fix the problem: https://firefox-ci-tc.services.mozilla.com/api/queue/v1/task/e66HwndkSWaiWqgmlEShWg/runs/0/artifacts/public/build/geckoview_example.apk

**Comment by xM8WVqaG** [comment]
Thanks for that @jamienicol, I've tested the 64 bit APKs

> This apk should be broken: https://firefox-ci-tc.services.mozilla.com/api/queue/v1/task/VYds3sRoSbq23ZdDjEBS7w/runs/0/artifacts/public/build/geckoview_example.apk

This APK appears to be broken - it loads blank webpages similarly to `>= 84.1.0`.

>This apk should hopefully fix the problem: https://firefox-ci-tc.services.mozilla.com/api/queue/v1/task/e66HwndkSWaiWqgmlEShWg/runs/0/artifacts/public/build/geckoview_example.apk

This APK appears to be fixed - it loads webpages correctly like previous versions.

**Comment by jamienicol** [comment]
That's brilliant, thank you for testing.

Sorry about the inconvenience this has caused. We were trying to use an OpenGL extension to improve performance, but there appears to be a driver bug affecting Adreno GPUs on Android 6. More recent versions of Android don't have this bug so we didn't notice during our testing.

**Comment by xM8WVqaG** [comment]
Thanks for the rapid turn around! I can confirm pages are loading correctly again in [84.1.2](https://github.com/mozilla-mobile/fenix/releases/tag/v84.1.2) (not yet available in the Play Store, but the Play Store is slow).

**Comment by abodea** [comment]
Verified as fixed on the following builds: **84.1.2 dot RC, 85.0.0.beta4** and **Focus 8.11.2 GV 84.**
With the following devices: **Google Pixel 4 XL (11), Samsung Galaxy S10+ (10), Nexus 4 (Android 5), Moto G (Android 10),  Xiaomi Redmi Note 8T (Android 9), Nexus 5 (Android 6.0.1).**
Due to very specific devices running Adreno 510, we tried to verify this bug on the closest devices we got.
I also tried with emulators, but no luck there finding one virtual device with Adreno 510.

Thank you @xM8WVqaG for the confirmation!
